src/app_model/types/_keys/_keybindings.py

```diff
@@ -136,10 +138,60 @@ class KeyBinding(BaseModel):
     the two keypress codes with a space. For example, 'Ctrl+K Ctrl+C'.
     """
 
-    parts: List[SimpleKeyBinding] = Field(..., min_items=1)
+    __root__: str
+    _parts: List[SimpleKeyBinding] = PrivateAttr()
+
+    @classmethod
+    def _parts_error(cls, exc: Exception) -> ValidationError:
+        return ValidationError(
+            errors=[ErrorWrapper(exc=exc, loc=("parts",))], model=cls
+        )
+
+    def __init__(self, **data: Any) -> None:
+        if "parts" in data:
+            parts = data.pop("parts")
+            if not isinstance(parts, List):
+                raise self._parts_error(ListError())
+            parts = [SimpleKeyBinding.validate(part) for part in parts]
+        elif "__root__" in data:
+            root = data["__root__"]
+            parts = [SimpleKeyBinding.from_str(part) for part in root.split()]
+        else:
+            raise self._parts_error(MissingError())
+
+        if len(parts) < 1:
+            raise self._parts_error(ListMinLengthError(limit_value=1))
+        data["__root__"] = " ".join(str(part) for part in parts)
+
+        super().__init__(**data)
+        self._parts = parts
+
+    @property
+    def parts(self) -> List[SimpleKeyBinding]:
+        """Key combinations that make up the overall key chord."""
+        return self._parts
+
+    @no_type_check
+    def __setattr__(self, key, val):
```

Comment:
Idea: do we need to support mutation of `KeyBinding`? This adds extra code to maintain, which is mostly just a repeat of `__init__` anyway.

I don't think performance is a serious concern here right? Especially since `parts` is likely to remain the only member of this thing anyway.

If we made this model frozen/immutable, then we could just derive the `parts` property (likely as a cached property) and maybe add a factory method like `KeyBinding.from_parts` to avoid some of the complexity in `__init__` too.